command/kv_delete.go

```diff
@@ -0,0 +1,178 @@
+package command
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+)
+
+// KVDeleteCommand is a Command implementation that is used to delete a key or
+// prefix of keys from the key-value store.
+type KVDeleteCommand struct {
+	Ui cli.Ui
+}
+
+func (c *KVDeleteCommand) Help() string {
+	helpText := `
+Usage: consul kv delete [options] KEY_OR_PREFIX
+
+  Removes the value from Consul's key-value store at the given path. If no
+  key exists at the path, no action is taken.
+
+  To delete the value for the key named "foo" in the key-value store:
+
+      $ consul kv delete foo
+
+  To delete all keys which start with "foo", specify the -recurse option:
+
+      $ consul kv delete -recurse foo
+
+  This will delete the keys named "foo", "food", and "foo/bar/zip" if they
+  existed.
+
+` + apiOptsText + `
+
+KV Delete Options:
+
+  -cas                    Perform a Check-And-Set operation. If this value is
+                          specified without -modify-index, the key will first be
+                          fetched and the resulting ModifyIndex will be used on
+                          the next query. The default value is false.
+
+  -modify-index=<int>     Unsigned integer representing the ModifyIndex of the
+                          key. This is often combined with the -cas flag, but it
+                          can be specified for any key. The default value is 0.
```

Comment:
Consul will ignore this on any other key operations (you can't manually set ModifyIndex) so I'd just change this to:

This is used in combination with the -cas flag.

You might even be able to just do -cas=<int> but that might not be very intuitive.